### PR TITLE
Windows build cleanup

### PR DESCRIPTION
**Pull request by uklotzde**
- Move defines out of source files into CMakeLists.txt to (fixes some warnings in our Windows build)
- Drop obsolete Windows XP support

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1949,6 +1956,9 @@ target_link_libraries(mixxx-lib PRIVATE ReplayGain)
 
 # Reverb
 add_library(Reverb STATIC EXCLUDE_FROM_ALL lib/reverb/Reverb.cc)
+if(MSVC)
+  target_compile_definitions(Reverb PUBLIC _USE_MATH_DEFINES)
```
Comment (Holzhaus) [inline CMakeLists.txt:1960]: Do we need this flags just for these libraries? If so, can we make these PRIVATE ?

Comment (uklotzde) [reply CMakeLists.txt:1960]: I will try to restrict or remove some of them. Trial and error.

**Comment by uklotzde** [comment]
Another failed test on Windows that never failed on AppVeyor. The GitHub runtime environment seems to cause unpredictable issues when running teh test.

**Comment by Holzhaus** [comment]
> Another failed test on Windows that never failed on AppVeyor. The GitHub runtime environment seems to cause unpredictable issues when running teh test.

I fear we have data races in the code on windows. Since Appveyor was much slower, me might not have noticed them. We should try to narrow down the cause by adding DEBUG_ASSERTs.

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you!

**Pull request merged**